test/Sdk.Analyzers.Tests/RegistrationExpectedInAspNetIntegrationTests.cs

```diff
@@ -0,0 +1,277 @@
+                    using Microsoft.Azure.Functions.Worker;
+                    using Microsoft.Extensions.DependencyInjection;
+                    using Microsoft.Extensions.Hosting;
+                    using Microsoft.Extensions.Logging;
+
+                    class Program
+                    {
+                        static void Main(string[] args)
+                        {
+                            var host = new HostBuilder()
+                                .ConfigureFunctionsWorkerDefaults()
+                                .Build();
+
+                            host.Run();
+                        }
+                    }
+                }
+
+                namespace AspNetIntegration
+                {
+                    using Microsoft.Azure.Functions.Worker;
+                    using Microsoft.Azure.Functions.Worker.Http;
+
+                    public class FunctionHttpTrigger
+                    {
+                        [Function(nameof(FunctionHttpTrigger))]
+                        public void Run([HttpTrigger(AuthorizationLevel.Anonymous, ""post"")] HttpRequestData req)
+                        {
+                        }
+                    }
+                }";
+
+            var test = new AnalyzerTest
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net60.WithPackages(ImmutableArray.Create(
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker", "1.19.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Sdk", "1.14.1"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs", "6.0.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore", "1.0.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Abstractions", "5.0.0"),
+                    new PackageIdentity("Microsoft.Extensions.Hosting.Abstractions", "6.0.0"),
+                    new PackageIdentity("Microsoft.Azure.Functions.Worker.Extensions.Http", "3.1.0")
+                    )),
+
+                TestCode = testCode
+            };
+
+            test.ExpectedDiagnostics.Add(Verify.Diagnostic()
+                            .WithSeverity(Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                            .WithSpan(2, 17, 22, 18)
+                            .WithArguments(ExpectedRegistrationMethod));
+
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task NotAspNetIntegration_Diagnostics_NotExpected()
+        {
+            string testCode = @"
+                using System;
+                using Microsoft.Extensions.DependencyInjection;
+                using Microsoft.Extensions.Hosting;
+
+                namespace SampleApp
+                {
+                    public class Program
+                    {
+                        public static void Main()
```

Comment:
Add another private or public method to this class in one of those tests.

Reply:
done